chunk_downloader.go

```diff
@@ -707,6 +707,12 @@ type ArrowBatch struct {
 	rowCount           int
 	scd                *snowflakeChunkDownloader
 	funcDownloadHelper func(context.Context, *snowflakeChunkDownloader, int) error
+	ctx                context.Context
```

Comment:
if it's an internal interface, have you updated all the call sites? If it's a public interface and we won't be able to enumerate all the call sites, we will need to make the extra parameter optional with a default value, right?